src/pages/index.js

```diff
@@ -40,89 +50,222 @@ const enableValidation = (config) => {
 
 enableValidation(config);
 
-//const editFormValidator = new FormValidator(config, profileEditForm);
-//editFormValidator.enableValidation();
-//const cardAddFormValidator = new FormValidator(config, cardAddForm);
-//cardAddFormValidator.enableValidation();
+/* -------------------------------------------------------------------------- */
+/*                                   Profile                                  */
+/* -------------------------------------------------------------------------- */
 /* -------------------------------- UserInfo -------------------------------- */
-const profileInfo = new UserInfo(".profile__name", ".profile__description");
-/* ------------------------------ PopupWithForm ----------------------------- */
+const profileInfo = new UserInfo(
+  ".profile__name",
+  ".profile__description",
+  ".profile__avatar"
+);
+
+api
+  .getUserInfo()
+  .then((res) => {
+    profileInfo.setUserInfo({ name: res.name, description: res.about });
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+/* ------------------------------ ProfilePopup ----------------------------- */
 const profileEditPopup = new PopupWithForm({
   popupSelector: "#profile-edit-modal",
   handleFormSubmit: handleProfileEditSubmit,
 });
 profileEditPopup.setEventListeners();
-const cardAddPopup = new PopupWithForm({
-  popupSelector: "#card-add-modal",
-  handleFormSubmit: handleCardAddSubmit,
-});
-cardAddPopup.setEventListeners();
-/* ----------------------------- PopupWithImage (Card)----------------------------- */
-const cardImagePopup = new PopupWithImage("#image-modal");
-cardImagePopup.setEventListeners();
-/* --------------------------------- Section -------------------------------- */
-const cardList = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
-  },
-  ".cards__list"
-);
-cardList.renderItems();
 
-/* -------------------------------------------------------------------------- */
-/*                                  Functions                                 */
-/* -------------------------------------------------------------------------- */
-/* --------------------------------- Profile -------------------------------- */
 function fillProfileForm() {
   const userData = profileInfo.getUserInfo();
   profileTitleInput.value = userData.name;
   profileDescriptionInput.value = userData.job;
 }
-/* ---------------------------------- Card ---------------------------------- */
 
-function handleImageClick(name, link) {
-  cardImagePopup.open(name, link);
+function handleProfileEditSubmit(userData) {
+  profileEditPopup.renderLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```